wordpress-mobile/WordPress-Android
# [Dependency Updates] Restructure Dependency Versions

Parent: #17551
Closes: #17554

This PR restructures and groups the dependency versions for the whole project. In addition to the more obvious benefits of doing so, this is done as a pre-work for versions catalog, but more so, for the dependency updates project that is about to start.

PS: Reviewing this PR should be done commit-by-commit, otherwise it will end-up becoming too overwhelming to review or miss something. Having said that, I understand that there are 100 commits included in this PR (so far). Thus, it might take some time to go through each and every one of them, but I thought it is better to be safe and do the commits step-by-step instead of batching them into multiple commits, potentially gaining some extra speed. Apologies in advance if there was a better way to do that and still make it easy to review this change. 🙏 

-----

ℹ️ This restructure of dependency versions is not updating any dependencies. As such, it is safe to assume that no changes are introduced to both, the `WordPress` and `Jetpack` apps.

For example see:
- [Wiremock] This ec64693a2fee6d491599dc72cfc89d50299f9342 commit that got reverted by this 0b94f7a6c8761753296dcbcd284ba6d9a81b7a0b commit and created duplicate `wiremock` and `wiremockMocks` versions.
- [AppCompat] This 794e43b4a0aaf4e47f7175efabb7e786ada5f358 commit that got reverted by this e5dc6c50f5eee22d73c8362f5d20ef301e6f9125 commit and created duplicate `androidxAppcompat` and `androidxAppcompatImageEditor` versions.

-----

Version Sections:
- `libs` (a.k.a. `automattic/wordpress`)
- `debug`
- `main`
- `test`
- `android test`
- `other`

Dependency Sections:
- `file`
- `project/compositeBuild`
- `automattic/wordpress` (a.k.a `libs`)
- `debugImplementation`
- `compileOnly/api/implementation/kapt`
- `testProject`
- `testImplementation/kaptTest`
- `androidTestProject`
- `androidTestImplementation/kaptAndroidTest`
- `detektPlugins/lintChecks/coreLibraryDesugaring`

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you really want to be thorough:
    - Check the latest dependency tree diff and verify that everything looks as expected (example [here](https://github.com/wordpress-mobile/WordPress-Android/pull/17535#issuecomment-1327273527)). PS: I am not sure why there is any diff there, that is, since no version got updated. Maybe this is because I changed the order of dependencies, not sure... 🤷 
    - Quickly smoke test both, the `WordPress` and `Jetpack` apps, and see if they work as expected.

-----

### Merge instructions

- [x] Make sure to thoroughly review and test this PR (see instructions above).
- [x] If everything looks good and confidence in merging this change is high, proceed.
- [x] Update PR with latest `trunk` (resolve any conflicts with `rebase`, not `merge`).
- [x] Remove `[Do Not Merge]` prefix from the PR title.

-----

## Regression Notes

1. Potential unintended areas of impact

- Unintended dependency updates due to wrong search-and-replace. 😅 
- Unintended transitive dependency updates due to placing dependencies in different order. 🤷 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
